modules/library/main/src/main/java/org/geotools/filter/function/FilterFunction_in.java

```diff
@@ -0,0 +1,81 @@
+/*
+ *    GeoTools - The Open Source Java GIS Toolkit
+ *    http://geotools.org
+ * 
+ *    (C) 2015, Open Source Geospatial Foundation (OSGeo)
+ *    
+ *    This library is free software; you can redistribute it and/or
+ *    modify it under the terms of the GNU Lesser General Public
+ *    License as published by the Free Software Foundation;
+ *    version 2.1 of the License.
+ *
+ *    This library is distributed in the hope that it will be useful,
+ *    but WITHOUT ANY WARRANTY; without even the implied warranty of
+ *    MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
+ *    Lesser General Public License for more details.
+ */
+
+package org.geotools.filter.function;
+
+import java.util.List;
+
+import org.geotools.filter.FunctionExpressionImpl;
+import org.opengis.filter.capability.FunctionName;
+import org.opengis.filter.expression.Expression;
+
+/**
+ * The function checks whether a candidate value is contained in an arbitrary long list
+ * of user provided values.
+ * 
+ * <p>
+ * If the candidate value is found, the function returns <code>true</code>;
+ * otherwise, it returns <code>false</code>.
+ * </p>
+ * 
+ * @author Stefano Costa, GeoSolutions
+ * 
+ * 
+ * 
+ * @source $URL$
+ *
+ */
+public class FilterFunction_in extends FunctionExpressionImpl {
```

Comment:
Just call it InFunction, the FilterFunction_ naming convention is from a code generator that Dave Blasby wrote.